### ISSUE TITLE
Unable to properly marshall data in iOS for type coordinates: interop.Pointer | interop.Reference<CLLocationCoordinate2D>

### ISSUE THREAD

**Issue by Tyler-V** [NativeScript/NativeScript#8450]
**Environment**
Provide version numbers for the following components (information can be retrieved by running `tns info` in your project folder or by inspecting the `package.json` of the project):
 - CLI: 6.4.0
 - Cross-platform modules:
 - Android Runtime: 6.4.1
 - iOS Runtime: 6.4.2
 - Plugin(s): 
implementation 'com.mapbox.mapboxsdk:mapbox-android-sdk:8.6.2'
pod 'Mapbox-iOS-SDK', '~> 5.6.1'

**Describe the bug**
When attempting to call methods in the mapbox sdk, I am unable to marshall the data into the correct format shown in the examples depicted in Mapbox's documentation for iOS.

In this [documentation provided:](https://docs.mapbox.com/ios/api/maps/5.7.0/Classes/MGLMapView.html#/c:objc(cs)MGLMapView(im)setVisibleCoordinates:count:edgePadding:direction:duration:animationTimingFunction:completionHandler:)

I am unable to create an object of type `coordinates: interop.Pointer | interop.Reference<CLLocationCoordinate2D>`

```
      const coordinates: CLLocationCoordinate2D[] = [];
      for (let latLng of latLngs) {
        const coordinate = CLLocationCoordinate2DMake(latLng.lng, latLng.lat);
        coordinates.push(coordinate);
      }

      const array: any = NSArray.arrayWithArray<CLLocationCoordinate2D>(coordinates);

      mapView.setVisibleCoordinatesCountEdgePaddingDirectionDurationAnimationTimingFunctionCompletionHandler(
        array,
        coordinates.length,
        insets,
        0,
        duration / 1000,
        CAMediaTimingFunction.functionWithName(kCAMediaTimingFunctionEaseInEaseOut),
        () => {
          resolve();
        }
      );
```

This calls the method without error, but what is passed in is an array of CLLocationCoordinate2D that is marshalled incorrectly. I know this because instead of the provided latLngs array taking me to a central part of the US which works on Android, it takes me off the coast of Africa.

**To Reproduce**
Try to create a coordinates object of type interop.Pointer | interop.Reference<CLLocationCoordinate2D>, inspect the values created and coordinates are not correct.

**Expected behavior**
I should be able to create an object of type interop.Pointer | interop.Reference<CLLocationCoordinate2D> where the value is properly mapped and doesn't result in an improper object.

**Sample project**
I would be more than happy to invite you into my plugin's project which has a branch already setup and a demo-angular app which will let you bootstrap and debug this issue with minimal effort to see for yourself. Please let me know otherwise this should be easily reproducible given the steps listed above.

**Comment by facetious** [comment]
The function you are using has the signature:
```objective-c
CLLocationCoordinate2D CLLocationCoordinate2DMake(CLLocationDegrees latitude, CLLocationDegrees longitude);
```

You have passed the arguments:
```typescript
CLLocationCoordinate2DMake(latLng.lng, latLng.lat);
```

Presumably, your `.lng` maps to a longitude and `.lat` maps to a latitude, which is the wrong order for the signature you're invoking.

**Comment by Tyler-V** [comment]
@facetious good catch but unfortunately that's residual code leftover on trying to make it display properly.

CLLocationCoordinate2DMake(latLng.lng, latLng.lat) = Just south of Nigeria over the ocean.
CLLocationCoordinate2DMake(latLng.lat, latLng.lng) = Further south west from Africa.

**Comment by NickIliev** [comment]
@Tyler-V see [this blog post](https://www.nativescript.org/blog/use-native-ios-static-libraries-and-js-modules-from-npm-with-latest-1.1-release) which pretty much covers the case with `CLLocationCoordinate2DMake`

>     GMSServices.provideAPIKey(APIKEY);
>  
>     console.log("Creating map view...");
>     camera = GMSCameraPosition.cameraWithLatitudeLongitudeZoom(-33.86, 151.20, 6);
>     mapView = GMSMapView.mapWithFrameCamera(CGRectZero, camera);
>  
>     console.log("Setting a marker...");
>     marker = GMSMarker.alloc().init();
>     // Note that in-line functions such as CLLocationCoordinate2DMake are not exported.
>     marker.position = { latitude: -33.86, longitude: 151.20 }
>     marker.title ="Sydney";
>     marker.snippet ="Australia";

**Comment by Tyler-V** [comment]
Hey @NickIliev thanks for the resource.

I don't think there is anything in that blog post that applies to my situation, sorry if my issue was unclear. I don't have any issues with creating a CLLocationCoordinate, that appears to be working just fine in other areas of my plugin.

The real issue at hand here is the marshalling of data and how to create the object that it is asking for.

I have an array of CLLocationCoordinates.

It wants the following:
`coordinates: interop.Pointer | interop.Reference<CLLocationCoordinate2D>, count: number`

As a workaround, I somehow cobbled together something that actually calls [this API method:](https://docs.mapbox.com/ios/api/maps/5.7.0/Classes/MGLMapView.html#/c:objc(cs)MGLMapView(im)setVisibleCoordinates:count:edgePadding:direction:duration:animationTimingFunction:completionHandler:)
```
      const coordinates: CLLocationCoordinate2D[] = [];
      for (let latLng of latLngs) {
        const coordinate = CLLocationCoordinate2DMake(latLng.lat, latLng.lng);
        coordinates.push(coordinate);
      }

      const array: any = NSArray.arrayWithArray<CLLocationCoordinate2D>(coordinates);

      mapView.setVisibleCoordinatesCountEdgePaddingDirectionDurationAnimationTimingFunctionCompletionHandler(
        array,
        coordinates.length,
        insets,
        0,
        duration / 1000,
        CAMediaTimingFunction.functionWithName(kCAMediaTimingFunctionEaseInEaseOut),
        () => {
          resolve();
        }
      );
```

However, it sends my map to southwest Africa instead of where the latLngs are (US)... swapping the lat/lngs sends me further south from Africa over the ocean.
